Bastion/main.go

```diff
@@ -0,0 +1,239 @@
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		err = createConnectors(connectors)
+		if err != nil {
+			log.Printf("Connectors creation failed: %v", err)
+			http.Error(w, http.StatusText(400), 400)
+			return
+		}
+		w.Write([]byte(fmt.Sprint("Connectors created")))
+	})
+
+	r.Post("/bastion/topics", func(w http.ResponseWriter, r *http.Request) {
+		var topics []models.Topic
+		err := parseJson(w, r, &topics)
+
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		err = createTopics(topics)
+
+		if err != nil {
+			log.Printf("Topics creation failed: %v", err)
+			http.Error(w, http.StatusText(400), 400)
+			return
+		}
+
+		w.Write([]byte(fmt.Sprint("Topics created")))
+	})
+
+	port := fmt.Sprintf(":%v", os.Getenv(portKey))
+	log.Printf("Listening at server: %v", port)
+	http.ListenAndServe(port, r)
+}
+
+func parseJson(w http.ResponseWriter, r *http.Request, dst interface{}) error {
+
+	r.Body = http.MaxBytesReader(w, r.Body, 1048576)
+
+	dec := json.NewDecoder(r.Body)
+	dec.DisallowUnknownFields()
+
+	// To improve error handling checkout https://www.alexedwards.net/blog/how-to-properly-parse-a-json-request-body
+	return dec.Decode(&dst)
+}
+
+func createDatabases(databases []models.Database) error {
+	dbpool, err := pgxpool.Connect(context.Background(), os.Getenv(databaseUrlKey))
+	if err != nil {
+		log.Printf("Unable to connect to database: %v\n\n", err)
+		os.Exit(1)
+	}
+
+	defer dbpool.Close()
+
+	for _, d := range databases {
+		createDb := "CREATE DATABASE \"" + d.Name + "\" OWNER " + d.Owner
+
+		_, err = dbpool.Exec(context.Background(), createDb)
+		if err != nil {
+			log.Printf("Create %v failed: %v\n", d.Name, err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createConnectors(connectors []models.Connector) error {
+	var eg errgroup.Group
+
+	log.Println("createConnectors: Starting workers")
+
+	for _, c := range connectors {
+		connector := c
+		eg.Go(func() error {
+			return createConnector(connector)
+		})
+	}
+
+	log.Println("createConnectors: Waiting for workers to finish")
+	err := eg.Wait()
+	return err
+}
+
+func createConnector(connector models.Connector) error {
+	b := new(bytes.Buffer)
+	err := json.NewEncoder(b).Encode(connector)
+	if err != nil {
+		return err
+	}
+
+	res, err := http.Post(debeziumHost, "application/json", b)
+
+	if err != nil {
+		return err
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode > 299 {
+		if res.Body != nil {
+			bodyBytes, err := io.ReadAll(res.Body)
+			if err != nil {
+				log.Fatal(err)
+			}
+			bodyString := string(bodyBytes)
+
+			return fmt.Errorf("debezium connector error with status: %v response: %v", res.StatusCode, bodyString)
+		}
+
+		return fmt.Errorf("debezium connector error with status: %v", res.StatusCode)
+	}
+
+	return err
+}
+
+func createTopics(topics []models.Topic) error {
+	adminClient, err := kafka.NewAdminClient(&kafka.ConfigMap{"bootstrap.servers": bootstrapServers})
```

Comment:
adminClient, err := kafka.NewAdminClient(&kafka.ConfigMap{"bootstrap.servers": bootstrapServers, "security.protocol": "SSL"}) to be able to connect using TLS on kafka brokers